Jenkinsfile

```diff
@@ -5,6 +5,9 @@ pipeline {
       image 'tarampampam/node:13-alpine'
     }
   }
+  environment {
+    HOME = '.'
```

Comment:
This is interesting... setting `HOME` to `.` means "home is the current directory". I wonder if that will have other unintended consequences? Can you let me know what error you are getting?

Reply:
The specific error is at the bottom of this. Sorry for pasting all of the output; I thought you might need the docker output for context. One note, setting the environment like this is actually shown in the Jenkinsfile in the Sonatype guide.

```
Branch indexing
 > git rev-parse --resolve-git-dir /var/lib/jenkins/caches/git-5541f82576bf18acd6df768b724d9903/.git # timeout=10
Setting origin to https://github.com/non-local/nexus-iq-for-scm-demo.git
 > git config remote.origin.url https://github.com/non-local/nexus-iq-for-scm-demo.git[](https://abradley-jenkins.sonatypedev.com:445/job/npm-example/job/main/3/console#)[](https://abradley-jenkins.sonatypedev.com:445/job/npm-example/job/main/3/console#) # timeout=10
Fetching origin...
Fetching upstream changes from origin
 > git --version # timeout=10
 > git --version # 'git version 2.32.0'
 > git config --get remote.origin.url # timeout=10
using GIT_ASKPASS to set credentials github
 > git fetch --tags --force --progress -- origin +refs/heads/*:refs/remotes/origin/* # timeout=10
Seen branch in repository origin/86ac60/react-dom/16.0.0-to-16.0.1
Seen branch in repository origin/main
Seen 2 remote branches
Obtained [Jenkins](https://abradley-jenkins.sonatypedev.com:445/computer/(built-in)/)[](https://abradley-jenkins.sonatypedev.com:445/job/npm-example/job/main/3/console#)[](https://abradley-jenkins.sonatypedev.com:445/job/npm-example/job/main/3/console#)[](https://abradley-jenkins.sonatypedev.com:445/job/npm-example/job/main/3/console#)[](https://abradley-jenkins.sonatypedev.com:445/job/npm-example/job/main/3/console#)file from c5d6ad2c9367b4aa1ec8e6f2634c5c5a3d7b787e
[Pipeline] Start of Pipeline
[Pipeline] node
Running on Jenkins in /var/lib/jenkins/workspace/npm-example_main
[Pipeline] {
[Pipeline] stage
[Pipeline] { (Declarative: Checkout SCM)
[Pipeline] checkout
Selected Git installation does not exist. Using Default
The recommended git tool is: NONE
using credential github
 > git rev-parse --resolve-git-dir /var/lib/jenkins/workspace/npm-example_main/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/non-local/nexus-iq-for-scm-demo.git # timeout=10
Fetching without tags
Fetching upstream changes from https://github.com/non-local/nexus-iq-for-scm-demo.git
 > git --version # timeout=10
 > git --version # 'git version 2.32.0'
using GIT_ASKPASS to set credentials github
 > git fetch --no-tags --force --progress -- https://github.com/non-local/nexus-iq-for-scm-demo.git[](https://abradley-jenkins.sonatypedev.com:445/job/npm-example/job/main/3/console#)[](https://abradley-jenkins.sonatypedev.com:445/job/npm-example/job/main/3/console#)[](https://abradley-jenkins.sonatypedev.com:445/job/npm-example/job/main/3/console#)[](https://abradley-jenkins.sonatypedev.com:445/job/npm-example/job/main/3/console#)[](https://abradley-jenkins.sonatypedev.com:445/job/npm-example/job/main/3/console#)[](https://abradley-jenkins.sonatypedev.com:445/job/npm-example/job/main/3/console#)[](https://abradley-jenkins.sonatypedev.com:445/job/npm-example/job/main/3/console#)[](https://abradley-jenkins.sonatypedev.com:445/job/npm-example/job/main/3/console#)[](https://abradley-jenkins.sonatypedev.com:445/job/npm-example/job/main/3/console#)[](https://abradley-jenkins.sonatypedev.com:445/job/npm-example/job/main/3/console#)[](https://abradley-jenkins.sonatypedev.com:445/job/npm-example/job/main/3/console#) +refs/heads/*:refs/remotes/origin/* # timeout=10
Checking out Revision c5d6ad2c9367b4aa1ec8e6f2634c5c5a3d7b787e (main)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f c5d6ad2c9367b4aa1ec8e6f2634c5c5a3d7b787e # timeout=10
 > git branch -a -v --no-abbrev # timeout=10
 > git branch -D main # timeout=10
 > git checkout -b main c5d6ad2c9367b4aa1ec8e6f2634c5c5a3d7b787e # timeout=10
Commit message: "Update Jenkinsfile"
 > git rev-list --no-walk 6bb9def0f20a93784254c70c97a192f0281efad7 # timeout=10
[Pipeline] }
[Pipeline] // stage
[Pipeline] withEnv
[Pipeline] {
[Pipeline] isUnix
[Pipeline] withEnv
[Pipeline] {
[Pipeline] sh
+ docker inspect -f . tarampampam/node:13-alpine
.
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] withDockerContainer
Jenkins does not seem to be running inside a container
$ docker run -t -d -u 994:990 -w /var/lib/jenkins/workspace/npm-example_main -v /var/lib/jenkins/workspace/npm-example_main:/var/lib/jenkins/workspace/npm-example_main:rw,z -v /var/lib/jenkins/workspace/npm-example_main@tmp:/var/lib/jenkins/workspace/npm-example_main@tmp:rw,z -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** tarampampam/node:13-alpine cat
$ docker top 85183f6d25a5c0bed6af37cb8e75a687d8329a7b07e280950811e06bbeb5c91b -eo pid,comm
[Pipeline] {
[Pipeline] stage
[Pipeline] { (Build)
[Pipeline] sh
+ npm install
npm ERR! correctMkdir failed to make directory /.npm/_locks
npm WARN app@1.0.0 No description
npm WARN app@1.0.0 No repository field.
npm WARN app@1.0.0 No license field.

npm ERR! code EACCES
npm ERR! syscall mkdir
npm ERR! path /.npm
npm ERR! errno -13
npm ERR! 
npm ERR! Your cache folder contains root-owned files, due to a bug in
npm ERR! previous versions of npm which has since been addressed.
npm ERR! 
npm ERR! To permanently fix this problem, please run:
npm ERR!   sudo chown -R 994:990 "/.npm"
[Pipeline] }
```

Reply:
Here's the [guide](https://guides.sonatype.com/iqserver/technical-guides/iq-for-scm/)